### PR TITLE
[feat]: [SEI-2562]: BE  SEI tenant creation on Harness and licensing

### PR DESCRIPTION
**Pull request by ashish-levelops**
## Describe your changes
https://harness.atlassian.net/wiki/spaces/LEV/pages/21417754907/SEI+on+Harness+User+Experience#SEI-tenant-creation-and-licensing--
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51571)

### CODE REVIEW

File: 930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseSummaryHelper.java [hunk 1]
```diff
@@ -182,6 +184,19 @@ public static LicensesWithSummaryDTO generateSummary(
           }
         };
         break;
+      case SEI:
+        licensesWithSummaryDTO = SEILicenseSummaryDTO.builder().build();
+        summaryHandler = (moduleLicenseDTO, summaryDTO, current) -> {
+          SEIModuleLicenseDTO temp = (SEIModuleLicenseDTO) moduleLicenseDTO;
+          SEILicenseSummaryDTO seiLicenseSummaryDTO = (SEILicenseSummaryDTO) summaryDTO;
+          if (current < temp.getExpiryTime()) {
+            if (temp.getNumberOfContributors() != null) {
```
Comment (pbarapatre10) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseSummaryHelper.java:193]: nit: can be effectively combined using && clause, thanks

File: 930-ng-license-manager/src/main/java/io/harness/licensing/entities/modules/SEIModuleLicense.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@OwnedBy(HarnessTeam.SEI)
+@Data
+@Builder
+@EqualsAndHashCode(callSuper = true)
+@StoreIn(DbAliases.NG_MANAGER)
+@Entity(value = "moduleLicenses", noClassnameStored = true)
+@Persistent
+@TypeAlias("io.harness.license.entities.module.SEIModuleLicense")
+public class SEIModuleLicense extends ModuleLicense {
+    private Integer numberOfContributors;
```
Comment (pbarapatre10) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/entities/modules/SEIModuleLicense.java:23]: numberOfContributers is going to be different from numberOfUsers in Harness?

**Review by pbarapatre10 — APPROVED** [review]
LGTM

File: 925-enforcement-service/src/main/resources/features/sei_features.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+moduleType: SEI
+clients:
+  - name: sei
+    clientConfig: seiServiceClientConfig
+    secretConfig: nextGenConfig.ngManagerServiceSecret
```
Comment (kapil-harness) [inline 925-enforcement-service/src/main/resources/features/sei_features.yml:5]: is SEI not a separate service? if yes, you should add its respective secret

File: 930-ng-license-manager/src/main/java/io/harness/licensing/entities/modules/SEIModuleLicense.java (hunk 2 above)
Comment (ashish-levelops) [reply 930-ng-license-manager/src/main/java/io/harness/licensing/entities/modules/SEIModuleLicense.java:23]: Yes, numberOfContributers is total users for whom harness users wants to see engg insights

File: 925-enforcement-service/src/main/resources/features/sei_features.yml (hunk 3 above)
Comment (ashish-levelops) [reply 925-enforcement-service/src/main/resources/features/sei_features.yml:5]: @kapil-harness SEI service does not communicate with any of the harness service and I don't think we will need to communicate anytime in near future, so I thought we could reuse ngManagerServiceSecret as adding new secret is unnecessary overhead.
Correct me if I am missing anything

File: 930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseSummaryHelper.java (hunk 1 above)
Comment (ashish-levelops) [reply 930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseSummaryHelper.java]: Done

**Comment by ashish-levelops** [comment]
trigger uts

**Comment by abhay084** [comment]
trigger messagecheck

**Comment by abhay084** [comment]
trigger messagecheck

**Comment by AnshuH** [comment]
trigger mdc

**Comment by ashish-levelops** [comment]
trigger messagecheck

**Comment by ashish-levelops** [comment]
trigger checkstylecodeformat

**Comment by ashish-levelops** [comment]
trigger messagecheck

**Comment by ashish-levelops** [comment]
trigger messagecheck

**Review by kapil-harness — APPROVED** [review]

**Comment by ashish-levelops** [comment]
trigger ut2

**Comment by ashish-levelops** [comment]
trigger ut5

**Comment by ashish-levelops** [comment]
trigger utall

**Comment by ashish-levelops** [comment]
trigger uts

**Comment by ashish-levelops** [comment]
trigger ut4

**Pull request closed without merging**